src/Base/Placement.cpp

```diff
@@ -209,3 +210,23 @@ Placement Placement::sclerp(const Placement& p0, const Placement& p1, double t,
     Placement trf = p0.inverse() * p1;
     return p0 * trf.pow(t, shorten);
 }
+
+void Placement::print(std::string name)
+{
+
+    Base::Vector3d pos = getPosition();
+    Base::Vector3d axis;
+    double angle;
+    Base::Rotation rot = getRotation();
+    rot.getRawValue(axis, angle);
+    Base::Console().Warning(
+        "placement %s : position (%.1f, %.1f, %.1f) - axis (%.1f, %.1f, %.1f) angle %.1f\n",
+        name.c_str(),
+        pos.x,
+        pos.y,
+        pos.z,
+        axis.x,
+        axis.y,
+        axis.z,
+        angle);
+}

```

Comment:
```suggestion
std::string Placement::toString()
{
    Base::Vector3d pos = getPosition();
    Base::Rotation rot = getRotation();
    
    Base::Vector3d axis;
    double angle;
    rot.getRawValue(axis, angle);
    
    return fmt::format(
        "position ({.1f}, {.1f}, {.1f}), axis ({.1f}, {.1f}, {.1f}), angle {.1f}\n",
        pos.x,
        pos.y,
        pos.z,
        axis.x,
        axis.y,
        axis.z,
        angle);
}
```

Then you can use it in more scenarios, with different error levels, with combine it with different strings, `qDebug()` etc.